packages/domain/src/use-cases/lifecycle/Bootstrap.ts

```diff
@@ -70,7 +71,9 @@ export class Bootstrap<T extends TransactionRequestModel> {
   private async startWithCredentials(wallet: Wallet) {
     this.activeWalletPresenter.presentActiveWallet(wallet);
 
-    await this.activeProfileLoader.loadActiveProfileByOwnerAddress(wallet.address);
+    const profiles = await this.profileGateway.getAllProfilesByOwnerAddress(wallet.address);
+
+    await this.activeProfileLoader.loadActiveProfileByHandle(profiles[0].handle);
```

Comment:
`ActiveProfileLoader` is a convenience object aimed at encapsulating some application specific domain logic around active profile management. This also helped isolate dependencies that are needed for this purpose (gateway and presenter) from the interactors that uses this helper object.

In this new configuration the `Bootstrap` interactor has a new reference to the `IProfileGateway`, retains a reference to the `ActiveProfileLoader` and uses the first to instruct the second on what profile to make as active. The second then retrieve the profile again from the `IProfileGateway`.

I would consider to put this responsibility back into the `ActiveProfileLoader` so that the data flow inefficiency disappaer.

But before doing that I would also observe that using the `profiles[0]` at bootstrap time is the cause of the bug @desfero mentioned here: https://github.com/lens-protocol/lens-sdk/pull/137#discussion_r1097123961

Let's fix that and optimise this once we have the full picture of the final data flow.

Reply:
Got it, makes sense and will fix!